crates/client/sync/src/fetch/fetchers.rs

```diff
@@ -173,14 +170,38 @@ async fn fetch_class_update(
     Ok(classes.into_iter().flatten().collect())
 }
 
-/// Downloads a class definition from the Starknet sequencer. Note that because
-/// of the current type hell this needs to be converted into a blockifier equivalent
-async fn fetch_class(
-    class_hash: Felt,
+/// This method is used to fetch a class definition from the sequencer gateway in it's raw format for an easier conversion.
+pub async fn raw_get_class_by_hash(
+    gateway_url: &str,
+    class_hash: &str,
     block_number: u64,
-    provider: &SequencerGatewayProvider,
-) -> Result<ContractClassData, ProviderError> {
-    let core_class = provider.get_class(BlockIdCore::Number(block_number), class_hash).await?;
+) -> Result<serde_json::Value, ProviderError> {
+    let client = Client::new();
+    let url = format!(
+        "{}/feeder_gateway/get_class_by_hash?classHash={}&blockNumber={}",
+        gateway_url, class_hash, block_number
+    );
+    let response = client.get(&url).send().await.map_err(|_| ProviderError::ArrayLengthMismatch)?;
```

Comment:
The current error handling for the HTTP response in the code is too generic and does not provide enough context about the actual error that occurred